src/Language/PureScript/TypeChecker.hs

```diff
@@ -516,7 +516,7 @@ typeCheckModule (Module _ _ _ _ Nothing) =
 typeCheckModule (Module ss coms mn decls (Just exps)) =
   warnAndRethrow (addHint (ErrorInModule mn)) $ do
     modify (\s -> s { checkCurrentModule = Just mn })
-    decls' <- typeCheckAll mn exps decls
+    decls' <- escalateWarningWhen unknownValueHintError $ typeCheckAll mn exps decls
```

Comment:
By escalating `UnknownValueHint` warnings into errors after the whole module is type checked, all `UnknownValueHint`s are collected before the `throwError` happens, fixing the multiple errors regression.